### PR TITLE
changes in login and register activity....

### PR DESCRIPTION
**Pull request by JesusArmandoM**

### CODE REVIEW

File: app/src/main/java/com/example/deakyu/fitnessapp/LoginActivity.kt [hunk 1]
```diff
@@ -115,7 +115,7 @@ class LoginActivity : AppCompatActivity(){
 
     private fun isPasswordValid(password: String): Boolean {
         //TODO: Replace this with your own logic
-        return password != null && password.length > 4
+        return password.length > 4
```
Comment (itzcoatl90) [inline app/src/main/java/com/example/deakyu/fitnessapp/LoginActivity.kt:118]: nice catch

File: app/src/main/java/com/example/deakyu/fitnessapp/RegisterActivity.kt [hunk 2]
```diff
@@ -31,16 +31,24 @@ class RegisterActivity : AppCompatActivity(){
         // Reset errors.
         email.error = null
         password.error = null
+        name.error = null
+        age.error = null
+        height.error = null
+        weight.error = null
```
Comment (itzcoatl90) [inline app/src/main/java/com/example/deakyu/fitnessapp/RegisterActivity.kt:37]: Are we assigning a null value to variables?   Is this a good practice in kotlin? does code line 33 and 32 comes by default created by Android?

File: app/src/main/java/com/example/deakyu/fitnessapp/RegisterActivity.kt [hunk 3]
```diff
@@ -31,16 +31,24 @@ class RegisterActivity : AppCompatActivity(){
...
+        val ageStr = age.text.toString()
+        val heightStr = height.text.toString()
+        val weightStr = weight.text.toString()
 
         var cancel = false
         var focusView: View? = null
 
         // Check for a valid password, if the user entered one.
-        if (!TextUtils.isEmpty(passwordStr) && !isPasswordValid(passwordStr)) {
+        if (TextUtils.isEmpty(passwordStr) && !isPasswordValid(passwordStr)) {
```
Comment (itzcoatl90) [inline app/src/main/java/com/example/deakyu/fitnessapp/RegisterActivity.kt:51]: Are we doing the same here and in LoginActivity? can we reuse the code?

File: app/src/main/java/com/example/deakyu/fitnessapp/RegisterActivity.kt [hunk 4]
```diff
@@ -81,7 +121,7 @@ class RegisterActivity : AppCompatActivity(){
 
     private fun isPasswordValid(password: String): Boolean {
```
Comment (itzcoatl90) [inline app/src/main/java/com/example/deakyu/fitnessapp/RegisterActivity.kt:122]: Same here, we are using the same lines of code coming from LoginActivity.... unless the Regular Expressions we are going to use are different, we should not do that

**Review by itzcoatl90 — CHANGES_REQUESTED** [review]
Please review

File: app/src/main/java/com/example/deakyu/fitnessapp/RegisterActivity.kt (hunk 4 above)
Comment (JesusArmandoM) [reply app/src/main/java/com/example/deakyu/fitnessapp/RegisterActivity.kt:122]: Yes we have to change this in order to not duplicate functionality, we can achieve this in the task that we need to do in order to authenticate to the user (with firebase) even more than that we need to create our viewmodel related with this activity (and change the layour to achieve mvvm)

File: app/src/main/java/com/example/deakyu/fitnessapp/RegisterActivity.kt (hunk 3 above)
Comment (JesusArmandoM) [reply app/src/main/java/com/example/deakyu/fitnessapp/RegisterActivity.kt:51]: Yes we have to change this in order to not duplicate functionality, we can achieve this in the task that we need to do in order to authenticate to the user (with firebase) even more than that we need to create our viewmodel related with this activity (and change the layour to achieve mvvm)

File: app/src/main/java/com/example/deakyu/fitnessapp/RegisterActivity.kt (hunk 2 above)
Comment (JesusArmandoM) [reply app/src/main/java/com/example/deakyu/fitnessapp/RegisterActivity.kt:37]: when you create a new project with kotlin and select the login (predesigned project), you can find this approach.

**Review by Deakyu — APPROVED** [review]

File: app/src/main/java/com/example/deakyu/fitnessapp/RegisterActivity.kt (hunk 2 above)
Comment (itzcoatl90) [reply app/src/main/java/com/example/deakyu/fitnessapp/RegisterActivity.kt:37]: ok

File: app/src/main/java/com/example/deakyu/fitnessapp/RegisterActivity.kt (hunk 3 above)
Comment (itzcoatl90) [reply app/src/main/java/com/example/deakyu/fitnessapp/RegisterActivity.kt:51]: What about imaging that this was still a pull request and that it did not proceed.... can we do that now? making this methods to be in an kind of login utility and not saving this to other ticket? please

File: app/src/main/java/com/example/deakyu/fitnessapp/RegisterActivity.kt (hunk 4 above)
Comment (itzcoatl90) [reply app/src/main/java/com/example/deakyu/fitnessapp/RegisterActivity.kt:122]: What about imaging that this was still a pull request and that it did not proceed.... can we do that now? making this methods to be in an kind of login utility and not saving this to other ticket? please

**Pull request merged**